### ISSUE TITLE
[BUG] Uncompilable haskell client with non-existent "AnyType"

### ISSUE THREAD

**Issue by adrianmay** [OpenAPITools/openapi-generator#13363]
#### Bug Report Checklist

- [ Yes] Have you provided a full/minimal spec to reproduce the issue?
- [ Tried, but it refused to download http://quorve.com/IP_RL_BSP_API-3.0.yaml ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ 6.0.1 ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ Y] Have you searched for related issues/PRs?
- [ Uncompilable haskell] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Makes uncompilable haskell

##### openapi-generator version

6.0.1

##### OpenAPI declaration file content or url

http://quorve.com/IP_RL_BSP_API-3.0.yaml

##### Generation Details

Installed openapi-generator from Arch repo.
Didn't change any configs.
Ran `openapi-generator generate -i IP_RL_BSP_API-3.0.yaml -g haskell-http-client`
Entered generated directory
`stack build`
```
/home/ad/TIKO/openapi/lib/IPRLBSP/Model.hs:465:43: error:
    Not in scope: type constructor or class ‘AnyType’
    |                    
465 |   , batchOperationResponseBody :: !(Maybe AnyType) -- ^ "body"
```
I can't find "AnyType" in Hoogle or the generated codebase.

##### Steps to reproduce

See above

##### Suggest a fix

Use a type that exists instead of AnyType

**Comment by JonathanLorimer** [comment]
Just want to second this issue, causing problems for me!

**Comment by wing328** [comment]
thanks for reporting the issue

what about mapping any type to `Dyanmic` from `Data.Dynamic` ?

ref: https://stackoverflow.com/a/6479498/677735

cc @jonschoning

**Comment by JonathanLorimer** [comment]
@wing328 I think `AnyType` here refers to any arbitrary json value, so [Value](https://hackage.haskell.org/package/aeson-2.2.3.0/docs/Data-Aeson-Types.html#t:Value) is actually a more accurate representation.

**Comment by jonschoning** [comment]
yes, it should be using `A.Value` here (arbitrary json value), somehow the existing type mapping isn't catching it.
